### ISSUE TITLE
[Bug] Invalid Player Data when starting singleplayer world in new 3.0 version

### ISSUE THREAD

**Issue by Tullemania** [iron431/Irons-Spells-n-Spellbooks#325]
### Observed behaviour

Start a world with new 3.0 version just gives me the Invalid player error

### Expected behaviour

I expected to get in to the world

### Steps to reproduce

Have the same mod list and start a singleplayer world. If needed I'l send it.

### Server Type

Single Player

### Crashlog

https://gist.github.com/Tullemania/f879774796f1a4958fb00151b038d06e

### Iron's Spells N Spellbooks version

2.30.2-3.0.0

### Forge version

1.20.1

### Other mods

I have 437 other mods and i don't know a way to list them all other then writing them out by hand

**Comment by iron431** [comment]
Is this the ATM9 pack?

**Comment by Tullemania** [comment]
This is in my own pack

**Comment by MrFastZombie** [comment]
I've had this issue when joining multiplayer too. I've compared mine and Tullemania's logs and we both have the following error:
<details>
<summary>Error from server latest.log</summary>
<br>

```
[28Feb2024 16:36:10.118] [Server thread/ERROR] [net.minecraft.server.network.ServerLoginPacketListenerImpl/]: Couldn't place player in world
java.lang.ClassCastException: class io.redspace.ironsspellbooks.loot.FurledMapLootFunction cannot be cast to class io.redspace.ironsspellbooks.loot.RandomizeRingEnhancementFunction (io.redspace.ironsspellbooks.loot.FurledMapLootFunction and io.redspace.ironsspellbooks.loot.RandomizeRingEnhancementFunction are in module irons_spellbooks@1.20.1-3.0.0 of loader 'TRANSFORMER' @7f4f432b)
	at io.redspace.ironsspellbooks.loot.RandomizeRingEnhancementFunction$Serializer.m_6170_(RandomizeRingEnhancementFunction.java:37) ~[irons_spellbooks-1.20.1-3.0.0.jar%23694!/:1.20.1-3.0.0]
	at net.minecraft.world.level.storage.loot.GsonAdapterFactory$JsonAdapter.serialize(GsonAdapterFactory.java:125) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.write(TreeTypeAdapter.java:88) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:70) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.internal.bind.ArrayTypeAdapter.write(ArrayTypeAdapter.java:104) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.Gson.toJson(Gson.java:825) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.Gson.toJsonTree(Gson.java:695) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.Gson.toJsonTree(Gson.java:672) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.serialize(TreeTypeAdapter.java:172) ~[gson-2.10.jar%2374!/:?]
	at net.minecraft.world.level.storage.loot.entries.LootPoolSingletonContainer$Serializer.m_7219_(LootPoolSingletonContainer.java:141) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.world.level.storage.loot.entries.LootItem$Serializer.m_7219_(LootItem.java:43) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.world.level.storage.loot.entries.LootItem$Serializer.m_7219_(LootItem.java:40) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer$Serializer.m_6170_(LootPoolEntryContainer.java:80) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.world.level.storage.loot.entries.LootPoolEntryContainer$Serializer.m_6170_(LootPoolEntryContainer.java:74) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.world.level.storage.loot.GsonAdapterFactory$JsonAdapter.serialize(GsonAdapterFactory.java:125) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.write(TreeTypeAdapter.java:88) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:70) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.internal.bind.ArrayTypeAdapter.write(ArrayTypeAdapter.java:104) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.Gson.toJson(Gson.java:825) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.Gson.toJsonTree(Gson.java:695) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.Gson.toJsonTree(Gson.java:672) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.serialize(TreeTypeAdapter.java:172) ~[gson-2.10.jar%2374!/:?]
	at net.minecraft.world.level.storage.loot.LootPool$Serializer.serialize(LootPool.java:206) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.world.level.storage.loot.LootPool$Serializer.serialize(LootPool.java:189) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.write(TreeTypeAdapter.java:88) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.write(TypeAdapterRuntimeTypeWrapper.java:70) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.write(CollectionTypeAdapterFactory.java:97) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.write(CollectionTypeAdapterFactory.java:61) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.Gson.toJson(Gson.java:825) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.Gson.toJsonTree(Gson.java:695) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.Gson.toJsonTree(Gson.java:672) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.serialize(TreeTypeAdapter.java:172) ~[gson-2.10.jar%2374!/:?]
	at net.minecraft.world.level.storage.loot.LootTable$Serializer.serialize(LootTable.java:329) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.world.level.storage.loot.LootTable$Serializer.serialize(LootTable.java:291) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at com.google.gson.internal.bind.TreeTypeAdapter.write(TreeTypeAdapter.java:88) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.Gson.toJson(Gson.java:825) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.Gson.toJsonTree(Gson.java:695) ~[gson-2.10.jar%2374!/:?]
	at com.google.gson.Gson.toJsonTree(Gson.java:672) ~[gson-2.10.jar%2374!/:?]
	at uk.me.desert_island.rer.RoughlyEnoughResources.sendLootToPlayers(RoughlyEnoughResources.java:53) ~[rer-2.9.0.jar%23791!/:?]
	at net.minecraft.server.players.PlayerList.handler$bcj000$roughlyenoughresources$onPlayerConnect(PlayerList.java:1952) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.server.players.PlayerList.m_11261_(PlayerList.java:262) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.server.network.ServerLoginPacketListenerImpl.m_143699_(ServerLoginPacketListenerImpl.java:139) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.server.network.ServerLoginPacketListenerImpl.m_10055_(ServerLoginPacketListenerImpl.java:126) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.server.network.ServerLoginPacketListenerImpl.redirect$zii000$fabric_networking_api_v1$handlePlayerJoin(ServerLoginPacketListenerImpl.java:563) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.server.network.ServerLoginPacketListenerImpl.m_9933_(ServerLoginPacketListenerImpl.java:70) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.network.Connection.m_129483_(Connection.java:263) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.server.network.ServerConnectionListener.m_9721_(ServerConnectionListener.java:142) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:907) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.server.dedicated.DedicatedServer.m_5703_(DedicatedServer.java:283) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:814) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:661) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:251) ~[server-1.20.1-20230612.114412-srg.jar%23875!/:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

</details>

The only other mod that is mentioned in the error appears to be Roughly Enough Resources:
`at uk.me.desert_island.rer.RoughlyEnoughResources.sendLootToPlayers(RoughlyEnoughResources.java:53) ~[rer-2.9.0.jar%23791!/:?]`
and sure enough this started happening in a clean instance w/ spellbooks once I've added RER.

**Comment by iron431** [comment]
I saw that error; I'm surprised nothing broke sooner and I don't know why REI has an issue with it or with 3.0.0, its been broken for a while. Its basically a typo though, and will be fixed in the next update coming soon

**Comment by Tullemania** [comment]
Thank you! :D

**Comment by iron431** [comment]
should be fixed in 3.0.1
